docs/src/main/asciidoc/getting-started-testing.adoc

```diff
@@ -1394,3 +1394,14 @@ For `@QuarkusIntegrationTest` tests that result in launcher the application as a
 This can be used by `QuarkusTestResourceLifecycleManager` that need to launch additional containers that the application will communicate with.
 ====
 
+== Troubleshooting
+
+=== Junit 5 parameterized tests failing on Java 16 and newer
+
+`@ParameterizedTest`s in JUnit 5 can fail with a `java.lang.IllegalStateException` (or
+`com.thoughtworks.xstream.converters.ConversionException` in older Quarkus versions) with a message `No converter available`
+for new-ish collection types like `java.util.Arrays$ArrayList` or other "JDK internal" collection types like
+`java.util.ImmutableCollections$List12`. The only workaround at this point is to add necessary `--add-opens` options.
```

Comment:
Isn't a better workaround to simply use `ArrayList`?

Reply:
@geoand we cannot provide custom serializers to XStream? Because if so, we could simply rewrite every list that is not an `ArrayList` to an `ArrayList` before delegating it to the existing serializer?
Wouldn't it work in our case? I don't think we care too much about the `List` type at the deserialization level?

Reply:
Very interesting idea! I tried it and it seems to work. https://github.com/quarkusio/quarkus/pull/24872 implements the idea.

What does everyone think?

Reply:
I think, it could work and would be much nicer (and make the doc change superfluous). For `Set` + `Map` it should probably use `LinkedHashSet` + `LinkedHashMap` though to retain order.